roles/result/tasks/fiod-samples.yml

```diff
@@ -0,0 +1,68 @@
+---
+
+
+# read_csv module was introduced in ansible version 2.8, and so can't be used at the time of
+# this PR, as there wasn't a stable tag for ansible-operator that was released with ansible
+# version 2.8.
+# - name: read the clat log file
+#   read_csv:
+#     path: "{{ item.path + '/' + 'fio_clat.1.log' }}"
+#     fieldnames: time_elapsed, value, data_direction, offset
+#     delimiter: ', '
+#   register: clat_log
+#
+# - name: template out the clat log file
+#   template:
+#     src: fio_clat_csv.json.j2
+#     dest: "{{results.path}}/{{workload.name}}-{{ uuid }}-fio-clat-output-{{item.time_elapsed}}.json"
+#   with_items: clat_log.list
+- name: set current job sequence
+  set_fact:
+    fio_jobsequence: "{{ item[0] }}"
+    fio_server_ip: "{{ item[1] }}"
+
+- name: Index log for clat
+  uri:
+    url: "http://{{ elasticsearch.server }}:{{ elasticsearch.port }}/{{ elasticsearch.index }}-clat/_doc"
+    method: POST
+    status_code: 201
+    body: "{{ lookup('template','fiod_clat.json.j2') }}"
+    body_format: json
+  loop: "{{ lookup('file', results.path + '/fiod-' + uuid + '/' + 'fio_clat.' + fio_jobsequence + '.log.' + fio_server_ip).splitlines() }}"
+
+
+- name: Index log for bw
+  uri:
+    url: "http://{{ elasticsearch.server }}:{{ elasticsearch.port }}/{{ elasticsearch.index }}-bw/_doc"
+    method: POST
+    status_code: 201
+    body: "{{ lookup('template','fiod_bw.json.j2') }}"
+    body_format: json
+  loop: "{{ lookup('file', results.path + '/fiod-' + uuid + '/' + 'fio_bw.' + fio_jobsequence + '.log.' + fio_server_ip).splitlines() }}"
+
+- name: Index log for iops
+  uri:
+    url: "http://{{ elasticsearch.server }}:{{ elasticsearch.port }}/{{ elasticsearch.index }}-iops/_doc"
+    method: POST
+    status_code: 201
+    body: "{{ lookup('template','fiod_iops.json.j2') }}"
+    body_format: json
+  loop: "{{ lookup('file', results.path + '/fiod-' + uuid + '/' + 'fio_iops.' + fio_jobsequence + '.log.' + fio_server_ip).splitlines() }}"
+
+- name: Index log for lat
+  uri:
+    url: "http://{{ elasticsearch.server }}:{{ elasticsearch.port }}/{{ elasticsearch.index }}-lat/_doc"
+    method: POST
+    status_code: 201
+    body: "{{ lookup('template','fiod_lat.json.j2') }}"
+    body_format: json
+  loop: "{{ lookup('file', results.path + '/fiod-' + uuid + '/' + 'fio_lat.' + fio_jobsequence + '.log.' + fio_server_ip).splitlines() }}"
+
+- name: Index log for slat
+  uri:
+    url: "http://{{ elasticsearch.server }}:{{ elasticsearch.port }}/{{ elasticsearch.index }}-slat/_doc"
+    method: POST
+    status_code: 201
+    body: "{{ lookup('template','fiod_slat.json.j2') }}"
+    body_format: json
+  loop: "{{ lookup('file', results.path + '/fiod-' + uuid + '/' + 'fio_slat.' + fio_jobsequence + '.log.' + fio_server_ip).splitlines() }}"
```

Comment:
Same thing with response codes and registering them.

Reply:
@aakarshg do you not want to store return codes

Reply:
So this particular task will fail if the response code is not 201. And it returns 201 only if its a success, and essentially it'd require a lot more logic to be added if we want to keep track of those that failed and those that passed, so we could retry later. This part of code is brittle for now, and i didn't know if it's going to help to put a lot of effort in it if this might go away later.